### PR TITLE
PP-2398 Added refund history capabilities

### PR DESCRIPTION
**Pull request by ckalista**
## WHAT
_A brief description of the pull request:_

- Leveraged `EclipseLink` extended support for tracking all changes made to `refunds` table. The `EclipseLink HistoryPolicy` has been configured on a `ClassDescriptor` via a custom `HistoryCustomizer` which one enabled with the appropriate `@Customizer` annotation on the `RefundEntity` will insert a row into the `refunds_history` table on every update to the primary `refunds` table to track the rows state. The history table also defines a start and end timestamp, so that the state of any row at any point in time can be determined.
- The `liquidbase` migration script was updates to add the `refunds_history` table in the database schema.
- Added an index on the `refunds_history` table to optimise performances while searching the history by `charge_id`.
- A POJO called `RefundHistory` was introduced as a data holder for refund history
   entries. It comes with appropriate SQL result set mappings to enable JPA native queries
   to map result sets direclty onto it.
- `RefundDao` was extended with a `searchHistoryByChargeId` method as its name implies allowing to search refunds by `chargeId` from the history.
- `ChargeEventsResource` has been reworked to incorporate refund events in its response messages. The structure of an event in the payload has been extended with a discriminator representative of the type of event which can be either: `PAYMENT` or `REFUND`. All events are listed in ascending order according to the `update` date.

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/ChargeEventsResourceITest.java [hunk 1]
```diff
@@ -137,41 +166,29 @@ public void shouldReturn404WhenIsChargeIdNotExists() {
...
+                        .aTestCharge()
+                        .withAmount(100L)
+                        .withTestAccount(testAccount)
+                        .withChargeStatus(CAPTURED);
     }
 
-    private String updateStatusTo(ChargeStatus chargeStatus) {
-        return toJson(ImmutableMap.of("new_status", chargeStatus.getValue()));
+    private DatabaseFixtures.TestChargeEvent createTestChargeEvent(DatabaseFixtures.TestCharge testCharge) {
+        return
```
Comment (maxcbc) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargeEventsResourceITest.java]: Does this need a new line between `return` and `with`?

File: src/test/java/uk/gov/pay/connector/it/resources/ChargeEventsResourceITest.java [hunk 2]
```diff
@@ -137,41 +166,29 @@ public void shouldReturn404WhenIsChargeIdNotExists() {
...
-                    JSON_RETURN_URL_KEY, returnUrl));
-        }
-
+    private DatabaseFixtures.TestCharge createTestCharge() {
+        DatabaseFixtures.TestAccount testAccount = withDatabaseTestHelper(databaseTestHelper)
+                .aTestAccount()
+                .withAccountId(Long.valueOf(accountId));
+
+        return
+                withDatabaseTestHelper(databaseTestHelper)
```
Comment (maxcbc) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargeEventsResourceITest.java]: Does this need a new line between return and with?

**Comment by ckalista** [comment]
Addressed code review comments from @maxcbc

File: src/main/java/uk/gov/pay/connector/model/domain/RefundHistory.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        setId(id);
+        setExternalId(externalId);
+        setAmount(amount);
+        setStatus(status);
+
+        ChargeEntity charge = new ChargeEntity();
+        charge.setId(chargeId);
+        setChargeEntity(charge);
+
+        setCreatedDate(new UTCDateTimeConverter().convertToEntityAttribute(createdDate));
```
Comment (rauligs) [inline src/main/java/uk/gov/pay/connector/model/domain/RefundHistory.java:22]: New instance of UTCDateTimeConverter per RefundHistory, can this be extracted? Same in HistoryStartDate and HistoryEndDate

**Review by rauligs — CHANGES_REQUESTED** [review]
Just one comment. It looks good!

**Review by rauligs — APPROVED** [review]

**Pull request merged**